### PR TITLE
My Notes Application

### LINKED ISSUE #81: Implement logic
MicroByte Title: Implement logic

Domain: Android Development (Java)

Prerequisites:
1. Need to know Basic Java, DSA, and Algorithm.
2. Knowledge of Android Studio.

Activities:
1. Layout Improvement in activity_main.xml
2. Coding Phase in the MainActivity.java

### PR DESCRIPTION
**Pull request by deepak-prajapatii**
# Closes #81 

Checklist: 
- [x] Added files or activities for learners.
- [x] Solutions are not given straight away.
- [x] Reference material for well understand for learners.
- [x] Comment the code in the file where the learner needs to write the code.
- [x] Added images so that learner compares their project after completing it.

### CODE REVIEW

**Comment by deepak-prajapatii** [comment]
Please review my PR.

File: My Notes Application/README.md
Comment (amoghrajesh) [inline My Notes Application/README.md]: Typo - advanced*

**Review by amoghrajesh — CHANGES_REQUESTED** [review]
Hey @deepak-prajapatii!
Sorry for the delay. 
Just went through your content and it seems little short to me.
Some suggestions from my end:

1. The content is a bit cut short, you need to add some more stuff for it to stretch about an hour.
2. Add a few more learner activities in the end - for example, this is just a plain design right, add an activity to maybe style to frame popup, adding some more features in the code, etc.
3. Try to breakdown the idea of the application interestingly by saying something like - the application comprises of a layout, part x, part y ... etc. (This will make a learner like it while doing it)
4. For XML reference, you have provided a link to a Udacity course, kindly change it to some blog or tutorial link online as nobody would want to do a course for a Byte. (Provide only reference links that are a need to know basis to the byte)

Apart from all these changes, good work!
Makes these changes asap and we shall re-review it. 
After making the changes, drop a comment that you are done, so we don't miss it out.

Thanks!

**Comment by deepak-prajapatii** [comment]
> Hey @deepak-prajapatii!
> Sorry for the delay.
> Just went through your content and it seems little short to me.
> Some suggestions from my end:
> 
> 1. The content is a bit cut short, you need to add some more stuff for it to stretch about an hour.
> 2. Add a few more learner activities in the end - for example, this is just a plain design right, add an activity to maybe style to frame popup, adding some more features in the code, etc.
> 3. Try to breakdown the idea of the application interestingly by saying something like - the application comprises of a layout, part x, part y ... etc. (This will make a learner like it while doing it)
> 4. For XML reference, you have provided a link to a Udacity course, kindly change it to some blog or tutorial link online as nobody would want to do a course for a Byte. (Provide only reference links that are a need to know basis to the byte)
> 
> Apart from all these changes, good work!
> Makes these changes asap and we shall re-review it.
> After making the changes, drop a comment that you are done, so we don't miss it out.
> 
> Thanks!

@amoghrajesh okay I understand

**Comment by deepak-prajapatii** [comment]
@amoghrajesh Changes had to be done according to you. Please review again.

**Review by amoghrajesh — CHANGES_REQUESTED** [review]
Hey @deepak-prajapatii!
Just reviewed it and it is a great improvement over the earlier version!
However, I observe some things that haven't been changed yet.

1. The typo on line 3 'advanced' has not been corrected yet.
2. My second point from the previous review has not been taken care of yet. You can add like a challenge in the end which is more engaging. For eg - You can ask the learner to construct a better layout for the frame. (This is just an example, consider asking on similar lines - provide question and references, not the solution)

Good improvement otherwise.
Hope to see these changed asap!

**Comment by deepak-prajapatii** [comment]
> Hey @deepak-prajapatii!
> Just reviewed it and it is a great improvement over the earlier version!
> However, I observe some things that haven't been changed yet.
> 
> 1. The typo on line 3 'advanced' has not been corrected yet.
> 2. My second point from the previous review has not been taken care of yet. You can add like a challenge in the end which is more engaging. For eg - You can ask the learner to construct a better layout for the frame. (This is just an example, consider asking on similar lines - provide question and references, not the solution)
> 
> Good improvement otherwise.
> Hope to see these changed asap!

@amoghrajesh Sorry I can't understand the typo for which you are talking about. Can you explain it?

**Comment by amoghrajesh** [comment]
@deepak-prajapatii Adding a screenshot:
image

**Comment by deepak-prajapatii** [comment]
@amoghrajesh I had corrected the typo that you are talked about earlier. I also add a challenge for the learners in the "micro challenge" section. Please review it again.

**Comment by deepak-prajapatii** [comment]
@amoghrajesh I had corrected the typo that you are talked about earlier. I also add a challenge for the learners in the "micro challenge" section. Please review it again.

**Review by amoghrajesh — CHANGES_REQUESTED** [review]
Ok, @deepak-prajapatii - 
I need you to add a little more to the learner challenge, what you have added currently is very basic. Add something more and provide more details about it, maybe links or so too which will help.

Kindly make this change

**Comment by deepak-prajapatii** [comment]
> Ok, @deepak-prajapatii - 
> I need you to add a little more to the learner challenge, what you have added currently is very basic. Add something more and provide more details about it, maybe links or so too which will help.
> 
> Kindly make this change

@amoghrajesh Sure

**Comment by deepak-prajapatii** [comment]
@amoghrajesh Done, check it again.

**Review by amoghrajesh — APPROVED** [review]
Hey @deepak-prajapatii!
Checked it. Good work on improving the content.
I really like the quick response from your end.
Approving it. Kindly wait to see what the other reviewers 😄 

Meanwhile, think of some learner issues that can be raised once this PR gets merged if possible.
Personally, I think you can raise a learner issue to solve the challenge at the end. (DO THIS ONLY AFTER MERGE). 
Also, have you thought how the learners will send their contributions in form of solution to the challenge? My proposal would be you create a folder called `Learner solutions` in `src` folder. Do this only after the merge.

Thanks for you contribution!

**Review by YourAKShaw — APPROVED** [review]
@deepak-prajapatii Thank you for your generous efforts and contributions. :heart_decoration: 

I see that you've been quite determined and have putting in your all. The we can use people like you in the community who is able to give his all.

**Just one quick reminder though, our reviewing process has been updated and we are getting more strict around the corners to ensure quality. Kindly checkout [this](https://github.com/Crio-Bytes/Miscellaneous/tree/main/COVID-19%20Visualization%20using%20Pandas%20and%20Matplotlib) MicroByte or even [this](https://github.com/Crio-Bytes/System-Design/tree/main/Exploring%20Content%20Delivery%20Network) for inspiration so that we can get a more premium quality of work from you in your next contribution.** 

**Also remember, quality > quantity.**

**Comment by YourAKShaw** [comment]
@deepak-prajapatii Kindly showcase your new MicroByte in our **#microbyte-showcase** channel over at our slack workspace using [this](https://github.com/Crio-Bytes/Demo-Repo/blob/master/Maintainer%20Resources/new-microbyte-showcase-template.md) template.

**Comment by YourAKShaw** [comment]
@all-contributors kindly add @deepak-prajapatii for:

- [x] code
- [x] doc
- [x] ideas
- [x] content

**Pull request merged**
